bst.py

```diff
@@ -82,4 +113,7 @@ def balance(self):
         than the left should return a negative value. An ideallyl-balanced tree
         should return 0.
         """
-        pass
+        if self._root:
+            return self._root.left_depth() - self._root.right_depth()
+        else:
+            return 0
```

Comment:
Given that the `_depth` attribute of every node is updated when a new node is inserted, You could make this a method of the node class, because each node should be able to answer it's own balance:

```python
class BSTNode(object):
    # ...
    return self.left_depth() - self.right_depth()

class BST(object):
    # ...
    def balance(self):
        return self._root.balance() if self._root else 0
```

```